extension/arquillian-drone-browserstack-extension/src/main/java/org/arquillian/drone/browserstack/extension/local/BrowserStackLocalRunner.java

```diff
@@ -0,0 +1,257 @@
+    private final File browserStackLocalFile =
+        new File(browserStackLocalDirectory.getPath() + File.separator + "BrowserStackLocal" + (PlatformUtils
+            .isWindows() ? ".exe" : ""));
+    private final String basicUrl = "https://www.browserstack.com/browserstack-local/";
+
+    private final CountDownLatch countDownLatch = new CountDownLatch(1);
+    private Process process = null;
+
+    private BrowserStackLocalRunner() {
+    }
+
+    /**
+     * Returns an instance of BrowserStackLocalRunner. If there has been already created, returns this one, otherwise
+     * creates and returns a new one - behaves like singleton
+     *
+     * @return An instance of BrowserStackLocalRunner
+     */
+    public static BrowserStackLocalRunner createBrowserStackLocalInstance() {
+        if (browserStackLocalRunner == null) {
+            browserStackLocalRunner = new BrowserStackLocalRunner();
+        }
+        return browserStackLocalRunner;
+    }
+
+    /**
+     * Indirectly runs BrowserStackLocal binary. In case that the binary has been already run, then does nothing.
+     *
+     * @param accessKey      An accessKey the binary should be ran with
+     * @param additionalArgs additional arguments
+     * @param localBinary    Path to a local binary of the BrowserStackLocal. If none, then it will be downloaded.
+     */
+    public void runBrowserStackLocal(String accessKey, String additionalArgs, String localBinary) {
+        if (process != null) {
+            return;
+        }
+        if (isEmpty(localBinary)) {
+            if (!browserStackLocalFile.exists()) {
+                prepareBrowserStackLocal();
+            }
+            runTheBinary(browserStackLocalFile, accessKey, additionalArgs);
+
+        } else {
+            runTheBinary(new File(localBinary), accessKey, additionalArgs);
+        }
+    }
+
+    /**
+     * Runs BrowserStackLocal binary. In case that the binary has been already run, then does nothing.
+     *
+     * @param binaryFile     A binary file to be run
+     * @param accessKey      An accessKey the binary should be ran with
+     * @param additionalArgs additional arguments
+     */
+    private void runTheBinary(File binaryFile, String accessKey, String additionalArgs) {
+        List<String> args = new ArrayList<String>();
+        args.add(binaryFile.getAbsolutePath());
+        args.add(accessKey);
+        if (!isEmpty(additionalArgs)) {
+            args.addAll(Arrays.asList(additionalArgs.split(" ")));
+        }
+        ProcessBuilder processBuilder = new ProcessBuilder().command(args);
+
+        try {
+            process = processBuilder.start();
+
+            final Reader reader = new Reader();
+            reader.start();
+            Runtime.getRuntime().addShutdownHook(new CloseChildProcess());
+            countDownLatch.await(20, TimeUnit.SECONDS);
+
+        } catch (UnsupportedEncodingException e) {
```

Comment:
are you still on Java 6? What about catching them all at once? Otherwise, just printing stacktrace is not the best thing to do ...